Tasks/Npm/npmtoolrunner.ts

```diff
@@ -23,13 +25,23 @@ export class NpmToolRunner extends tr.ToolRunner {
     public exec(options?: tr.IExecOptions): Q.Promise<number> {
         options = this._prepareNpmEnvironment(options) as tr.IExecOptions;
 
-        return super.exec(options);
+        return super.exec(options).catch((reason: any) => {
+            return this._printDebugLog(this._getDebugLogPath(options)).then((value: void): number => {
+                throw reason;
+            });
+        });
     }
 
     public execSync(options?: tr.IExecSyncOptions): tr.IExecSyncResult {
         options = this._prepareNpmEnvironment(options);
 
-        return super.execSync(options);
+        const execResult = super.execSync(options);
+        if (execResult.code !== 0) {
```

Comment:
I've checked the npm code and from what I can tell it will write to the debug log on every nonzero exit code. The only exception I could find was passing a bad command, which we can tell from npm's output.